service/src/main/java/bio/terra/pipelines/dependencies/cbas/CbasClient.java

```diff
@@ -0,0 +1,45 @@
+package bio.terra.pipelines.dependencies.cbas;
+
+import bio.terra.cbas.api.MethodsApi;
+import bio.terra.cbas.api.PublicApi;
+import bio.terra.cbas.api.RunSetsApi;
+import bio.terra.cbas.api.RunsApi;
+import bio.terra.cbas.client.ApiClient;
+import bio.terra.pipelines.app.configuration.external.CbasConfiguration;
+import org.springframework.stereotype.Component;
+
+@Component
+public class CbasClient {
+
+  private final CbasConfiguration cbasConfiguration;
+
+  public CbasClient(CbasConfiguration cbasConfiguration) {
+    this.cbasConfiguration = cbasConfiguration;
+  }
+
+  protected ApiClient getApiClient(String cbasBaseUri, String accessToken) {
+    ApiClient apiClient = new ApiClient().setBasePath(cbasBaseUri);
+    apiClient.addDefaultHeader("Authorization", "Bearer " + accessToken);
+    // By closing the connection after each request, we avoid the problem of the open connection
+    // being force-closed ungracefully by the Azure Relay/Listener infrastructure:
+    apiClient.addDefaultHeader("Connection", "close");
+    apiClient.setDebugging(cbasConfiguration.debugApiLogging());
+    return apiClient;
+  }
+
+  PublicApi publicApi(String cbasBaseUri, String accessToken) {
```

Comment:
it'd be helpful to have a brief comment for each of these describing what functionality each is used for